File: uke2/opg207-studenter-og-quiz-score/Student.java
class Student {
    private String navn;
    private int totSkore;
    private int antallQuizDeltatt;
    public Student(String name, int totScore, int numQuizDeltatt) {
	navn = name;
	totSkore = totScore;
	antallQuizDeltatt = numQuizDeltatt;
    }

    public String getName() {
	return navn;
    }

    public void leggTilQuizScore(int score){
	antallQuizDeltatt += 1;
	totSkore = totSkore + score;
    }

    public int hentTotalScore(){
	return totSkore;
    }
    
    public double hentGjennomsnittligScore(){
	return totSkore/((double) antallQuizDeltatt);
    }
}

File: uke2/opg207-studenter-og-quiz-score/c++/Student.cpp
#include "Student.h"

Student::Student(String name, int totalScore, int antallQuizDeltatt) :
  m_name(name), m_totalScore(totalScore), m_antallQuizDeltatt(antallQuizDeltatt) { }

void Student::leggTilQuizScore(int score) {
  m_totalScore += score;
  m_antallQuizDeltatt++;
}

String Student::getName() {
  return m_name;
}

int Student::hentTotalScore() {
  return m_totalScore;
}

double Student::hentGjennomsnittligScore() {
  return (double)m_totalScore / m_antallQuizDeltatt;
}
